kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/shared/KieServerExecutor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.server.integrationtests.shared;
+
+import org.kie.server.integrationtests.shared.basetests.KieServerBaseIntegrationTest;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import org.jboss.resteasy.plugins.server.tjws.TJWSEmbeddedJaxrsServer;
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.api.KieServerEnvironment;
+import org.kie.server.integrationtests.config.JacksonRestEasyTestConfig;
+import org.kie.server.integrationtests.config.TestConfig;
+import org.kie.server.remote.rest.common.resource.KieServerRestImpl;
+import org.kie.server.services.api.KieServerExtension;
+import org.kie.server.services.api.SupportedTransports;
+import org.kie.server.services.impl.KieServerImpl;
+
+public class KieServerExecutor {
+
+    protected TJWSEmbeddedJaxrsServer server;
+
+    // Need to hold kie server instance because we need to manually handle startup/shutdown behavior defined in
+    // context listener org.kie.server.services.Bootstrap. Embedded server doesn't support ServletContextListeners.
+    private KieServerImpl kieServer;
+
+    private static SimpleDateFormat serverIdSuffixDateFormat = new SimpleDateFormat("yyyy-MM-DD-HHmmss_SSS");
+
+    public void startKieServer() {
```

Comment:
Could you maybe decompose the method? It doesn't just start a Kie Server, it also configures various things.